src/CMakeLists.txt

```diff
@@ -512,6 +512,9 @@ add_subdirectory(os)
 add_subdirectory(osd)
 
 set(ceph_osd_srcs
+  # Link the Object Class API implementation directly as intermediary
+  # static library (like libosd.a) nullifies the effect of `-rdynamic`.
+  objclass/class_api.cc
```

Comment:
Linking the objclass API directly now – just like in the case of crimson-osd. Otherwise [`-rdynamic we already pass`](https://github.com/ceph/ceph/blob/master/src/CMakeLists.txt#L49) had no effect.